impeller/aiks/canvas.cc

```diff
@@ -502,6 +504,19 @@ Picture Canvas::EndRecordingAsPicture() {
   Picture picture;
   picture.pass = std::move(base_pass_);
 
+  std::vector<SkRect> rects;
+  picture.pass->IterateAllEntities([&rects](const Entity& entity) {
```

Comment:
This is showing up as being pretty hot. This might need to be fixed to see a larger difference.

Reply:
Perhaps we could append to the Rtree instead of temp storage? I think that is how it was designed to work.

Reply:
The current RTree impl expects to get an array of `SkRect` objects. We could probably refactor it to make it append.

But this should be happening on the UI thread now.

Reply:
How would that help anything? The rects have to be accumulated at some point, you are just pushing it from your own code into code that you don't own - it won't actually save anything.

You still have to do the iteration no matter how you create the RTree. The bounds still have to be accumulated into a tree. If Rtree is going to accumulate the bounds for you before it makes them into a tree then accumulation is still happening, it's just not in "your" code. If you want RTree to rebuild the tree on every append then not only will it have to still accumulate the bounds, but you will also have to add rebuilding the tree overhead on each rect.

Meanwhile, the dl_aiks_canvas was already creating an RTree accumulator that does *all of this for you* plus it will give you the bounds at the end and then the class wasn't even using it...

Reply:
We could just ask the rtree for the bounds.

Reply:
Correct, but I though this discussion was about creating the RTree, not using it to get the bounds...?